app/controllers/authentication.js

```diff
@@ -233,6 +233,49 @@ class AuthenticationController {
     };
 
     /*
+      Retrieve the user's github profile email.
+    */
+    const getEmail = (accessToken, headers, profile, next) => {
+      logger.debug(addPrefix('fetching user email information from github.'));
+
+      if (profile.email) {
+        next();
+        return;
+      }
+
+      logger.verbose(addPrefix('requesting profile emaile information.'));
+      const userEmailUrl = `${userApiUrl}/emails`;
+      request.get({url: userEmailUrl, qs: accessToken, headers: headers, json: true}, (error, response, emails) => {
+        logger.verbose(addPrefix(`response from: ${userEmailUrl} received.`));
+
+        // Process error if one happened
+        if (error) {
+          logger.warn(addPrefix(`getEmail error, failed to fetch user profile emails from url: ${userEmailUrl}.`));
+          return next({status: 500, msg: error.toString()});
+        }
+
+        // Make sure response was a 200 success
+        if (response.statusCode !== 200) {
+          logger.warn(addPrefix(`bad profile emails response with status code: ${response.statusCode}.`));
+          return next({
+            status: 409,
+            msg: `Could not fetch github profile. Response body: ${JSON.stringify(response.body)}`
+          });
+        }
+
+        // Make sure profile contains an email
+        if (!_.isArray(emails) || emails.length === 0) {
+          logger.warn(addPrefix('could not find email from fetched profile.'));
+          return next({status: 409, msg: 'Could not find email address from profile.'});
```

Comment:
Message is not wrong... but it could describe the situation a bit more accurately, eg. "Could not fetch emails from github profile, received invalid response body: ${emails}."

Also 409 Conflict does not sound right. If received emails is not an array, either github changed the user API or server mishandled the response. In both cases the **server** is at fault, so 500 code would be more fitting.